test/psych/test_string.rb

```diff
@@ -223,6 +223,13 @@ def test_string_with_ivars
       assert_equal ivar, food.instance_variable_get(:@we_built_this_city)
     end
 
+    def test_string_matching_date
+      assert_separately(%w[-r ./lib/psych], <<~RUBY)
```

Comment:
```suggestion
      assert_separately(%w[-I lib -r psych], <<~RUBY)
```

I think you need to pass `-I` otherwise it risks loading the extension from stdlib.